extension/asapauthextension/config.go

```diff
@@ -0,0 +1,64 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package asapauthextension // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/asapauthextension"
+
+import (
+	"errors"
+
+	"go.opentelemetry.io/collector/config"
+	"go.uber.org/multierr"
+)
+
+var (
+	errNoKeyIDProvided      = errors.New("no key id provided in asapclient configuration")
+	errNoTTLProvided        = errors.New("no valid ttl was provided in asapclient configuration")
+	errNoIssuerProvided     = errors.New("no issuer provided in asapclient configuration")
+	errNoAudienceProvided   = errors.New("no audience provided in asapclient configuration")
+	errNoPrivateKeyProvided = errors.New("no private key provided in asapclient configuration")
+)
+
+type Config struct {
+	config.ExtensionSettings `mapstructure:",squash"`
+
+	KeyID string `mapstructure:"key_id"`
+
+	TTL int `mapstructure:"ttl_seconds"`
```

Comment:
Instead of int, you can have it as `time.Duration` directly, so that your users can configure a TTL like `90s`.

Reply:
I did miss that, I agree it is a good change to have :)

Reply:
Yeah I agree it's a better way to do it. All done!